gui/src/lib/gui-theme.ts

```diff
@@ -0,0 +1,34 @@
+import { writable } from 'svelte/store';
+
+export const guiThemeLocalStorageKey = 'io.deref.exo/gui-theme';
+
+function createGuiTheme() {
+  const ls = localStorage.getItem(guiThemeLocalStorageKey) || 'auto';
+
+  const { subscribe, set } = writable(ls);
+
+  const localSyncedSet = (s: string) => {
+    set(s);
+    localStorage.setItem(guiThemeLocalStorageKey, s);
+  };
+
+  return {
+    subscribe,
+    auto: () => localSyncedSet('auto'),
+    light: () => localSyncedSet('light'),
+    dark: () => localSyncedSet('dark'),
+    black: () => localSyncedSet('black'),
+  };
+}
+
+export const guiTheme = createGuiTheme();
+
+export const setGuiTheme = (theme: string) => {
```

Comment:
I was really confused by this name. It's not really "setGuiTheme" so much as it is "renderGuiTheme" or "applyGuiTheme". Aside, "gui theme" seems redundant - why not just "theme"?

Everywhere you call `auto()` or `light()` or similar, you then immediately call `setGuiTheme`. Could that be automatic as part of `localSyncedSet`?

Actually, moreover, this probably doesn't need to be here at all. The guiTheme export is already subscribable, so Layout should add a reactive block that re-runs each time the theme changes.